R/utils.R

```diff
@@ -437,18 +442,122 @@ rot_x <- function(angle=30, hjust = 1, vjust = NULL, vertical = FALSE, ...) {
     }
     angle <- 90
     vjust <- 0.5
+  } else {
+    if(is.character(hjust)) {
+      abort("hjust must be numeric or NULL in this case.")
+    }
   }
   x <- element_text(angle = angle, hjust = hjust, vjust = vjust, ...)
   theme(axis.text.x=x)
 }
 
 ##' @rdname rot_x
 ##' @export
-rot_y <- function(angle=30, hjust = 1, vjust = NULL,...) {
-  y <- element_text(angle = angle, hjust = hjust, vjust = vjust, ...)
+rot_y <- function(angle=30, hjust = 1, vjust = NULL, vertical = FALSE, ...) {
+  if(isTRUE(vertical)) {
+    if(is.character(vjust)) {
+      vjust <- match.arg(vjust, c("left", "right"))
+      if(vjust=="left") vjust <- 1
+      if(vjust=="right") vjust <- 0
+    } else {
+      if(missing(vjust)) vjust <- 1
+    }
+    angle <- 90
+    hjust <- 0.5
+  } else {
+    if(is.character(vjust)) {
+      abort("`vjust` must be numeric or NULL in this case.")
+    }
+  }
+  y <- element_text(angle = angle, hjust = hjust, vjust = vjust,, ...)
   theme(axis.text.y=y)
 }
 
+
+.rotxy <- function(x, ...) UseMethod(".rotxy")
+#' @export
+.rotxy.gg <- function(x, axis = "x", ...) {
+  if(axis=="x") {
+    x <- x + rot_x(...)
+  } else {
+    x <- x + rot_y(...)
+  }
+  x
+}
+#' @export
+.rotxy.patchwork <- function(x, axis = "x", ...) {
+  stopifnot(requireNamespace("patchwork", quietly = TRUE))
+  if(axis=="x") {
+    x <- x & rot_x(...)
+  } else {
+    x <- x & rot_y(...)
+  }
+  x
+}
+#' @export
+.rotxy.default <- function(x, ...) {
+  stop("invalid object; can only rotate gg or patchwork objects.")
+}
+
+#' Rotate axis tick marks in a list of plots
+#'
+#' Pass in a list of gg or patchwork objects and rotate tick marks on x or y
+#' axes.
+#'
+#' @param x a named list of gg or patchwork objects.
+#' @param at a character vector of list names to rotate.
+#' @param re a regular expression for selecting names to be used as `at`.
+#' @param axis which axis to rotate.
+#' @param ... additional arguments passed to [rot_x()] or [rot_y()].
+#'
+#' @details
+#' Note that all plots in the list need to be named. If
+#'
+#' @examples
+#' data <- pmplots_data_id()
+#'
+#' co <- c("STUDYc", "CPc", "RF")
+#' etas <- paste0("ETA", 1:3)
+#'
+#' x <- eta_cat(data, x = co, y = etas)
+#' names(x)
+#'
+#' x <- rot_at(x, at = "ETA1vRF", angle = 35)
+#' x$ETA1vRF
+#'
+#' x <- rot_at(x, re = "RF", vertical = TRUE)
+#' x$ETA2vRF
+#' x$ETA3vRF
+#'
+#' @md
+#' @export
+rot_at <- function(x, at = names(x), re = NULL, axis = c("x", "y"), ...) {
+  if(!is.list(x) || is.ggplot(x) || inherits(x, "patchwork")) {
```

Comment:
I'm wondering if people will be tempted to pass a `gg` or `patchwork` object; trying to handle that up front.

Reply:
Not worried about this so much now; if people use `rot_xy()`, dispatch will figure that out.